gap.go

```diff
@@ -11,6 +11,9 @@ var (
 	errAdvertisementPacketTooBig = errors.New("bluetooth: advertisement packet overflows")
 )
 
+// default connection timeout
+const DEFAULT_CONNECTION_TIMEOUT time.Duration = 10 * time.Second
```

Comment:
This 10 second const came from the hardcoded value in the macos adapter connect implementation. I have surfaced it in the general `gap.go` function so it can be used by other implementations if required later.

Reply:
Good points. I have moved it back to `gap_darwin.go` and renamed it to `defaultConnectionTimeout`.